app/controllers/contracts_controller.rb

```diff
@@ -1,26 +1,35 @@
 class ContractsController < ApplicationController
 
   def create
-    @accepted_quote = Quote.find(params[:id])
-    @customer_request = @accepted_quote.customer_request
-    @accepted_quote.update(accepted: true)
-    if Contract.create(quote_id: @accepted_quote.id, customer_request_id: @customer_request.id)
-      @customer_request.quotes.each do |quote|
-        if quote.accepted
-          CompanyMailer.accept_email(quote).deliver_now
-        else
-          CompanyMailer.decline_email(quote).deliver_now
-          @accepted_quote.update(accepted: false)
+    accepted_quote = Quote.find(params[:id])
+    customer_request = accepted_quote.customer_request
+    if current_customer.customer_requests.include?(customer_request)
+      if Contract.create(
+        quote_id: accepted_quote.id,
+        customer_request_id: customer_request.id
+      )
+        accepted_quote.update(accepted: true)
+        customer_request.quotes.each do |quote|
+          if quote.accepted
+            CompanyMailer.accept_email(quote).deliver_now
+          else
+            CompanyMailer.decline_email(quote).deliver_now
+            quote.update(accepted: false)
+          end
         end
+      else
+        redirect_to "/quotes"
       end
-    else
-      redirect_to "/quotes"
     end
   end
 
   def show
-    @contract = Contract.find_by(params[:id])
-    render "show.html.erb"
+    @contract = Contract.find(params[:id])
```

Comment:
We probably want to protect this better. The way it's written, this would let anyone view a contract that doesn't belong to them.